core/templates/pages/topic-editor-page/editor-tab/topic-editor-tab.directive.ts

```diff
@@ -132,6 +133,59 @@ angular.module('oppia').directive('topicEditorTab', [
             $scope.topicNameExists = false;
             $scope.topicUrlFragmentExists = false;
             $scope.hostname = WindowRef.nativeWindow.location.hostname;
+            $scope.availableSkillsForDiagnosticTest = (
+              $scope.topic.getSkillSummariesThatAreNotInDiagnosticTest());
+            $scope.selectedSkillsForDiagnosticTest = (
+              $scope.topic.getDiagnosticTestSkillSummaries());
+            $scope.isDiagnosticTestSkillsDropdownPresent = false;
+            $scope.isSelectedSkillsForDiagnosticTestEmpty = true;
+            $scope.selectedSkillForDiagnosticTest;
+
+            $scope.presentDiagnosticTestSkillDropdown = function() {
+              $scope.isDiagnosticTestSkillsDropdownPresent = true;
+            };
+
+            $scope.removeSkillDropdownForDiagnosticTest = function() {
+              $scope.isDiagnosticTestSkillsDropdownPresent = false;
+            };
+
+            $scope.addSkillForDiagnosticTest = function(skillToAdd) {
+              if (skillToAdd === null) {
```

Comment:
I really don't understand this case. Why can it even happen?

Reply:
The reason I have added a null check is that on selecting a diagnostic test skill from the dropdown the ngChange value changes from the skill description to null after selection because of which some errors were observed.
I have added a video that will help in understanding the problem, PTAL.

https://user-images.githubusercontent.com/57531197/180633219-6fd24dfa-2024-44d9-a3d6-3f1e7f0d17bc.mp4

Reply:
Hm, some thoughts:

- Firstly, the UI for this is a bit weird, you shouldn't need to click outside the area for the skill to be "saved". Can you add a save button on the right perhaps?
- The best advice I have is to follow how other dropdowns are done in the codebase -- just follow their paradigms exactly. You can use select2-dropdown or select. But basically look for something similar in the codebase and copy its functionality and "how it does things" as much as possible; I don't think other dropdowns need this "null" exception or run into this null issue.